DOCSP-47797 Add VS and AVS to indexes page (#462)
mongodb/docs-golang: source/fundamentals/indexes.txt

   :end-before: end-create-vector-search
   :dedent:

The following example creates an Atlas Search index on the ``plot`` field of the
``sample_mflix.movies`` collection:

.. literalinclude:: /includes/fundamentals/code-snippets/indexes/atlasVectorSearch.go
   :language: go
   :start-after: start-create-atlas-search
   :end-before: end-create-atlas-search
   :dedent:

List a Search Index
```````````````````

The following example lists the details of the specified search index:

.. literalinclude:: /includes/fundamentals/code-snippets/indexes/atlasVectorSearch.go
   :language: go
   :start-after: start-list-index
   :end-before: end-list-index
   :dedent:

Update Search Indexes
`````````````````````

You can update an Atlas Search or Vector Search index by specifying the name of the index 
and the new index definition.

The following example updates a vector search index by providing the name of the index and
a new index definition:

.. literalinclude:: /includes/fundamentals/code-snippets/indexes/atlasVectorSearch.go
   :language: go
   :start-after: start-update-index
   :end-before: end-update-index
   :dedent:

Delete Search Indexes
`````````````````````

> The following example deletes a search index with the specified name:

Reviewer: "a search index" would refer to both Atlas Search and Atlas Vector Search. If we'd like to say Atlas Search, then we should mention Atlas Vector Search as well, since the example applies to both.
```suggestion
The following example deletes an Atlas Search index or an Atlas Vector Search index with the specified name:
```
or
```suggestion
The following example deletes an Atlas Search or Atlas Vector Search index with the specified name:
```

an index definition to the ``createOne()`` method.

The following example creates an Atlas Vector Search index on the ``plot_embedding`` 
field in the ``sample_mflix.embedded_movies`` collection:

.. literalinclude:: /includes/fundamentals/code-snippets/indexes/atlasVectorSearch.go
   :language: go
   :start-after: start-create-vector-search
   :end-before: end-create-vector-search
   :dedent:

The following example creates an Atlas Search index on the ``plot`` field of the
``sample_mflix.movies`` collection:

.. literalinclude:: /includes/fundamentals/code-snippets/indexes/atlasVectorSearch.go
   :language: go
   :start-after: start-create-atlas-search
   :end-before: end-create-atlas-search
   :dedent:

List a Search Index
```````````````````

The following example lists the details of the specified search index:

.. literalinclude:: /includes/fundamentals/code-snippets/indexes/atlasVectorSearch.go
   :language: go
   :start-after: start-list-index
   :end-before: end-list-index
   :dedent:

Update Search Indexes
`````````````````````

You can update an Atlas Search or Atlas Vector Search index by specifying the name of the index 
and the new index definition.

> The following example updates an Atlas Vector Search index by providing the name of the index and

Author: I don't think showing an Atlas search definition is necessary here. The focus for this example is on the update method, which is used the same for both AS and AVS.

a new index definition:

.. literalinclude:: /includes/fundamentals/code-snippets/indexes/atlasVectorSearch.go
   :language: go
   :start-after: start-update-index
   :end-before: end-update-index
   :dedent:

Delete Search Indexes
`````````````````````

> The following example deletes an Atlas Search index with the specified name:

Author: Added that example applies to both

.. literalinclude:: /includes/fundamentals/code-snippets/indexes/atlasVectorSearch.go
   :language: go
   :start-after: start-delete-index
   :end-before: end-delete-index
   :dedent:

.. _golang-clustered-indexes:

> Clustered Indexes

Reviewer: Thanks!

The following sections contain code examples that demonstrate how to manage Atlas 
Search and Atlas Vector Search indexes.

Create a Search Index
`````````````````````

You can create an Atlas Search or an Atlas Vector Search index by providing
an index definition to the ``SearchIndexView.CreateOne()`` method.

The following example creates an Atlas Search index on the ``plot`` field of the
``sample_mflix.movies`` collection:

.. literalinclude:: /includes/fundamentals/code-snippets/indexes/atlasVectorSearch.go
   :language: go
   :start-after: start-create-atlas-search
   :end-before: end-create-atlas-search
   :dedent:

The following example creates an Atlas Vector Search index on the ``plot_embedding`` 
field in the ``sample_mflix.embedded_movies`` collection:

.. literalinclude:: /includes/fundamentals/code-snippets/indexes/atlasVectorSearch.go
   :language: go
   :start-after: start-create-vector-search
   :end-before: end-create-vector-search
   :dedent:

List a Search Index
```````````````````

You can use the ``SearchIndexView.List()`` method to list an Atlas Search or Atlas 
Vector Search index by specifying the name of the index.

> The following example lists the details of the specified search index:

Reviewer: Should this be updated to coincide with whatever we choose for the "Delete a Search Index" section? Similar situation

API Documentation
~~~~~~~~~~~~~~~~~

To learn more about the methods discussed in this

> guide, as well as related ones, see the following API Documentation:

Reviewer: ```suggestion
guide and related ones, see the following API Documentation:
```
to resolve the check failure

> - `DropOne() <{+api+}/mongo#IndexView.DropOne>`__

Author: Updated copy and moved CreateOne() to be next to DropOne()
